### PR TITLE
Fix many spelling mistakes

### PR DESCRIPTION
**Pull request by codehorseman**
Signed-off-by: codehorseman <cricis@yeah.net>

### CODE REVIEW

**Comment by oli-obk** [comment]
r? @oli-obk 

are you using a specific tool that we could use to automate this kind of change in the future?

https://github.com/rust-lang/rust/pull/94947 seems to have found some of the same typos. I would like to avoid regularly merging spelling PRs if we can automate it instead.

File: compiler/rustc_ty_utils/src/needs_drop.rs [hunk 1]
```diff
@@ -181,12 +181,12 @@ enum DtorType {
     /// "significant" / "insignificant".
     Insignificant,
 
-    /// Type has a `Drop` implentation.
+    /// Type has a `Drop` implantation.
```
Comment (VirrageS) [inline compiler/rustc_ty_utils/src/needs_drop.rs:184]: I think this should be
```suggestion
    /// Type has a `Drop` implementation.
```

File: compiler/rustc_ty_utils/src/needs_drop.rs [hunk 2]
```diff
@@ -181,12 +181,12 @@ enum DtorType {
...
     Significant,
 }
 
 // This is a helper function for `adt_drop_tys` and `adt_significant_drop_tys`.
-// Depending on the implentation of `adt_has_dtor`, it is used to check if the
+// Depending on the implantation of `adt_has_dtor`, it is used to check if the
```
Comment (VirrageS) [inline compiler/rustc_ty_utils/src/needs_drop.rs:189]: Same here
```suggestion
// Depending on the implementation of `adt_has_dtor`, it is used to check if the
```

Comment (codehorseman) [reply compiler/rustc_ty_utils/src/needs_drop.rs:189]: Thanks

**Comment by codehorseman** [comment]
> ☔ The latest upstream changes (presumably #94966) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

Done

**Comment by codehorseman** [comment]
> 

Yes,As this issue said, it can be integrated into CI. I use a tool called [misspell](https://github.com/client9/misspell)

**Comment by codehorseman** [comment]
> > are you using a specific tool that we could use to automate this kind of change in the future?
> > #94947 seems to have found some of the same typos. I would like to avoid regularly merging spelling PRs if we can automate it instead.

Yes,As this issue said, it can be integrated into CI. I use a tool called [misspell](https://github.com/client9/misspell)

**Pull request merged**